sound/soc/sof/sof-audio.h

```diff
@@ -47,11 +47,17 @@
 #define SOF_DAI_CLK_INTEL_SSP_MCLK	0
 #define SOF_DAI_CLK_INTEL_SSP_BCLK	1
 
+/* sof_widget_op definitions cover two types of abstractions:
+ * 1) sof_ipc_tplg_widget_ops
+ * 2) sof_ipc_tplg_ops
+ */
 enum sof_widget_op {
-	SOF_WIDGET_PREPARE,
-	SOF_WIDGET_SETUP,
-	SOF_WIDGET_FREE,
-	SOF_WIDGET_UNPREPARE,
+	SOF_IPC_TPLG_WIDGET_SETUP, /* added for consistency, handled in topology loading only */
+	SOF_IPC_TPLG_WIDGET_PREPARE,
+	SOF_IPC_TPLG_SETUP,
+	SOF_IPC_TPLG_FREE,
+	SOF_IPC_TPLG_WIDGET_UNPREPARE,
+	SOF_IPC_TPLG_WIDGET_FREE, /* added for consistency, handled in topology unloading only */
```

Comment:
I'm not sure if this is cleaner (for me).
```c
enum sof_widget_op {

	SOF_WIDGET_IPC_SETUP, /* added for consistency, handled in topology loading only */
	SOF_WIDGET_IPC_PREPARE,
	SOF_TPLG_WIDGET_SETUP,
	SOF_TPLG_WIDGET_FREE,
	SOF_WIDGET_IPC_UNPREPARE,
	SOF_WIDGET_IPC_FREE, /* added for consistency, handled in topology unloading only */
};
```
might be better to articulate the different level of these ops?

Reply:
Yes @ujfalusi, what you suggested is what I implemented earlier, but then had a change of mind. 
it's a trade-off between what the phases mean and how they are tied to ops. I chose to use the latter, so that the phases can be mapped to ops with no ambiguity, i.e.SOD_IPC_TPLG_WIDGET_PREPARE is linked to sof_ipc_tplg_widget_ops::prepare and SOF_IPC_TPLG_SETUP to sof_ipc_tplg_ops::widget_setup

That said, as usual, naming is the hardest part of coding, so I'll go with the flow on the consensus. @ranj063 @ujfalusi @bardliao please vote!

Reply:
from other point of view, the original definitions were actions that we do on the widgets, set it up, prepare, etc and they did not meant to cover the exact callback path and location.

I'm not sure which is better.

Reply:
the problem is that we have two SETUP phases (one implicit and done during topology parsing, and the other done when doing the actual IPC).

Reply:
Right, and I think it is confusing to add the `SOF_IPC_TPLG_WIDGET_SETUP` / `SOF_IPC_TPLG_WIDGET_FREE` in the sof_widget_ops as they are not used in code and while there is a comment, it does not help, I think.
And to add to the confusion the two setup does slightly different action under the same umbrella word of 'setup', it might have been better to use different names for sof_ipc_tplg_widget_ops::ipc_setp/ipc_free, like ipc_create/ipc_destroy or something, Or ipc_init/ipc_deinit.